packages/http/src/mocker/index.ts

```diff
@@ -41,20 +51,52 @@ const mock: IPrismComponents<IHttpOperation, IHttpRequest, IHttpResponse, IMockH
 
       return config;
     }),
-    chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-    chain(result => assembleResponse(result, payloadGenerator))
+    Reader.chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
+    Reader.chain(result => assembleResponse(result, payloadGenerator)),
+    Reader.chain(response =>
+      withLogger(logger =>
+        pipe(
+          response,
+          Either.map(r => runCallbacks({ resource, request: input.data, response: r })(logger)),
+          Either.chain(() => response)
+        )
+      )
+    )
   );
 };
 
+function runCallbacks({
+  resource,
+  request,
+  response,
+}: {
+  resource: IHttpOperation;
+  request: IHttpRequest;
+  response: IHttpResponse;
+}) {
+  return withLogger(logger =>
+    pipe(
+      Option.fromNullable(resource.callbacks),
+      Option.alt<IHttpCallbackOperation[]>(() => Option.some([])),
```

Comment:
I would move this check outside of this function and not call it at all in case the user is not able to provide the callback

Reply:
I don't recall why it is there, maybe some leftover. Thanks & fixed.